### PR TITLE
OD-293 [Fix] Add only required security rules

### PR DESCRIPTION
**Pull request by ivandevupp**
@sofiiakvasnevska
## Issue
OD-293 https://weboo.atlassian.net/browse/OD-293
## Description
After clicking on `Configure security rules' security rules are not added when feature that requires them are disabled
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cYlr638pbi
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -708,27 +708,24 @@ var DynamicLists = (function() {
...
-        }
-
-        var accessRuleIndex = -1;
-
-        defaultRule.type = [type];
-
-        accessRules.forEach(function(item, index) {
-          var typeIndex = item.type.indexOf(type);
+      for (var type in options) {
+        if (options.hasOwnProperty(type)) {
```
Comment (tonytlwu) [inline js/interface.js:712]: Why did you change the logic to the other way round?

```suggestion
        if (!options.hasOwnProperty(type)) {
          continue;
        }
```

Comment (ivandevupp) [reply js/interface.js:712]: There was:
```suggestion
        if (!options[type]) {
          continue;
        }
```
And it skipped needed iterations when ``options[type] === false`` 
I added `if (options.hasOwnProperty(type)) {` after creating PR because of Codacy error, linter didn't highlight it (version ^1.0.5)

I think there are no difference between `hasOwnProp` and `hasOwnProp + continue`, in both cases iteration will be skipped in a first row after condition evaluation, should I change this?

Comment (tonytlwu) [reply js/interface.js:712]: Yes please change it. Fliplet's coding standard references the Airbnb JavaScript style guide, with [this specific rule](https://github.com/airbnb/javascript/blob/master/README.md#blocks--no-else-return). The same logic applies to `continue` as well cc @sofiiakvasnevska The Fliplet coding standard has been updated to specify this, please make sure all developers are aware.

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Medium
Affected: Studio users configuring LFD components
Verify: Data source security is correctly configured

Squash and merge: Use PR title
